### PR TITLE
Continue to use same .receive() method on disconnection

### LINKED ISSUE #7273: On transient connection failures, original `.receive()` call stops.
When a transient failure occurs in the connection, the connection is revived, but the original `.receive(String partitionId, EventPosition position)` stops pushing out events. Consumers need to perform another `.receive()`.

### PR DESCRIPTION
**Pull request by conniey**
* Adds AmqpReceiveLinkProcessor to fetch new links when one dies.
* Fixes logging for ActiveToken provider to give the resource rather than a full scope

Fixes #7273

### CODE REVIEW

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubPartitionAsyncConsumer.java [hunk 1]
```diff
@@ -3,127 +3,68 @@
...
         this.consumerGroup = consumerGroup;
         this.partitionId = partitionId;
-        this.emitterProcessor = EmitterProcessor.create(prefetchCount, false);
+        this.emitterProcessor = amqpReceiveLinkProcessor
+            .map(message -> onMessageReceived(message))
+            .doOnNext(event -> {
+                // Keep track of the last position so if the link goes down, we don't start from the original location.
+                final Long offset = event.getData().getOffset();
+                if (offset != null) {
+                    currentEventPosition.set(EventPosition.fromOffset(offset));
```
Comment (srnagar) [inline sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubPartitionAsyncConsumer.java]: Would it be more efficient if we just kept track of offset (long value) and convert to EventPosition only when the link goes down to restore the link? It prevents creating a lot of EventPosition objects.

Comment (conniey) [reply sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubPartitionAsyncConsumer.java]: Makes sense. I'll update it.

**Pull request merged**